pkg/util/extsort/disk_sorter.go

```diff
@@ -1378,7 +1392,12 @@ func (w *diskSorterWriter) flush() error {
 	slices.SortFunc(w.kvs, func(a, b keyValue) int {
 		return bytes.Compare(a.key, b.key)
 	})
+	var lastKey []byte
 	for _, kv := range w.kvs {
+		if bytes.Equal(lastKey, kv.key) {
+			continue
```

Comment:
why we have to check lastKey now?

Reply:
The stable pebbledb requires all keys are sorted and deduped when construct a SST file.